src/__swaps__/screens/Swap/hooks/useSwapInputsController.ts

```diff
@@ -144,16 +147,15 @@ export function useSwapInputsController({
 
   const formattedInputNativeValue = useDerivedValue(() => {
     if ((inputMethod.value === 'slider' && percentageToSwap.value === 0) || !inputValues.value.inputNativeValue) {
-      return '$0.00';
+      return convertAmountToNativeDisplayWorklet(0, currentCurrency);
     }
 
-    const nativeValue = `$${Number(inputValues.value.inputNativeValue).toLocaleString('en-US', {
-      useGrouping: true,
-      minimumFractionDigits: 2,
-      maximumFractionDigits: 2,
-    })}`;
+    const inputValue =
```

Comment:
`inputNativeValue` should always be the raw unformatted version of the native value since the formatted version is the job of this `formattedInputValue` derived value in the first place.
however, I see an issue that we are introducing formatting in the initial value which we shouldn't be doing on line 51.

we should fix that, make sure `convertAmountToNativeDisplayWorklet` supports both numbers and strings, and then just be able to pass in the `inputNativeValue` without having to do these extra steps here.

Reply:
nice!